src/App.tsx

```diff
@@ -1,42 +1,89 @@
-import { UserSelector } from './components/UserSelector';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
+import { Post, User } from './types';
+import { getPosts, getUsers } from './api/dataFromServer';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const [loadingPosts, setLoadingPosts] = useState(false);
+  const [usersPosts, setUsersPost] = useState<Post[]>([]);
+  const [loadPostError, setLoadPostError] = useState(false);
+
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  const handleSelectedUser = async (user: User) => {
+    setSelectedPost(null);
+    setSelectedUser(user);
+    setLoadingPosts(true);
+    try {
+      const posts = await getPosts(user.id);
+
+      setUsersPost(posts);
+    } catch {
+      setLoadPostError(true);
+    } finally {
+      setLoadingPosts(false);
+    }
+  };
+
+  const conditionShowPost = !loadingPosts && selectedUser && !loadPostError;
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  onChangeSelectedUser={handleSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
-
-                <Loader />
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {loadingPosts && <Loader />}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {loadPostError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <PostsList />
+                {!usersPosts.length && conditionShowPost && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
+                {Boolean(usersPosts.length) && conditionShowPost && (
```

Comment:
More readable construction
```suggestion
                {usersPosts.length > 0 && conditionShowPost && (
```